### PR TITLE
Universal Profiling: on-prem GA additions

### PR DESCRIPTION
**Pull request by inge4pres**
## Description

Remove the beta banner in the on-perm doc, cover areas of Universal Profiling documentation to allow on-perm users to operate the backend.

* host-agent telemetry
* programmatic setup

### Documentation sets edited in this PR

_Check all that apply._

- [X] Stateful (`docs/en/observability/*`)
- [ ] Serverless (`docs/en/serverless/*`)
- [ ] Integrations Developer Guide (`docs/en/integrations/*`)
- [ ] None of the above

### Related issue

closes https://github.com/elastic/prodfiler/issues/5222 and elastic/prodfiler/issues/5319

## Checklist

- [ ] Product/Engineering Review
- [X] Writer Review

### Follow-up tasks

_Select one._

* This PR does _not_ need to be ported to another doc set because:
  - [X] The concepts in this PR only apply to one doc set (serverless _or_ stateful)
  - [ ] The PR contains edits to both doc sets (serverless _and_ stateful)
* This PR needs to be ported to another doc set:
  - [ ] Port to stateful docs: \<link to PR or tracking issue>
  - [ ] Port to serverless docs: \<link to PR or tracking issue>

### CODE REVIEW

File: docs/en/observability/profiling-get-started.asciidoc [hunk 1]
```diff
@@ -81,6 +81,30 @@ NOTE: To configure data ingestion, you need elevated privileges, typically the `
 
 If you're upgrading from a previous version with Universal Profiling enabled, see the <<profiling-upgrade,upgrade guide>>.
 
+[discrete]
+[[profiling-configure-data-ingestion-programmatic]]
+=== Programmatic configuration
+
+If you prefer to configure data ingestion programmatically, you can use a Kibana API call.
+The API call can be performed either via the "Dev Tools" console in Kibana or any standalone HTTP client (like `curl` or `wget`).
+In both cases, the API call will have tobe executed with the `elastic` user credentials to have the necessary permissions.
```
Comment (bmorelli25) [inline docs/en/observability/profiling-get-started.asciidoc:90]:
```suggestion
This call can be made either through the "Dev Tools" console in Kibana or with any standalone HTTP client (such as `curl` or `wget`).
In both cases, the API call must be executed using the `elastic` user credentials to ensure the necessary permissions.
```

File: docs/en/observability/profiling-get-started.asciidoc [hunk 2]
```diff
@@ -81,6 +81,30 @@ NOTE: To configure data ingestion, you need elevated privileges, typically the `
...
+[[profiling-configure-data-ingestion-programmatic]]
+=== Programmatic configuration
+
+If you prefer to configure data ingestion programmatically, you can use a Kibana API call.
+The API call can be performed either via the "Dev Tools" console in Kibana or any standalone HTTP client (like `curl` or `wget`).
+In both cases, the API call will have tobe executed with the `elastic` user credentials to have the necessary permissions.
+
+A successful API call will return a `202 Accepted` response with an empty body.
+
+To configure data ingestion from the console, go to *Dev Tools* from the navigation menu and execute the following snippet:
```
Comment (bmorelli25) [inline docs/en/observability/profiling-get-started.asciidoc:94]:
```suggestion
To configure data ingestion from the console, go to *Dev Tools* in the navigation menu and run the following command:

```

File: docs/en/observability/profiling-get-started.asciidoc [hunk 3]
```diff
@@ -81,6 +81,30 @@ NOTE: To configure data ingestion, you need elevated privileges, typically the `
...
+[source,console]
+----
+POST kbn:/internal/profiling/setup/es_resources
+{}
+----
+
+To configure data ingestion programmatically using a standalone HTTP client (e.g. `curl`), execute the following command:
```
Comment (bmorelli25) [inline docs/en/observability/profiling-get-started.asciidoc:101]:
```suggestion
To configure data ingestion programmatically using a standalone HTTP client (e.g., `curl`), run the following command:

```

File: docs/en/observability/profiling-self-managed-ops.asciidoc [hunk 4]
```diff
@@ -172,15 +172,81 @@ This endpoint is configured through the application's `host` configuration optio
 
 For example, if the collector is configured with the default value `host: 0.0.0.0:8260`, you can check the health of the application by running `curl -i localhost:8260/live` and `curl -i localhost:8260/ready`.
 
+[discrete]
+[[profiling-scaling-external-telemetry]]
+== Profiling agent telemetry data
+
+The Universal Profiling collector receives from profiling agent a set of telemetry data that is used to debug the operations performed by the agent,
+as well as collecting product usage statistics.
+We use the data collected to understand the demographics of the profiled machines and to perform investigations when problems are reported to our customer service.
```
Comment (bmorelli25) [inline docs/en/observability/profiling-self-managed-ops.asciidoc:181]:
```suggestion
The Universal Profiling collector receives telemetry data from profiling agents to help debug agent operations and gather product usage statistics.
This data enables us to understand the demographics of profiled machines and aids in investigations when customers report issues.
```

File: docs/en/observability/profiling-self-managed-ops.asciidoc [hunk 5]
```diff
@@ -172,15 +172,81 @@ This endpoint is configured through the application's `host` configuration optio
...
+
+By default, telemetry data collected by all profiling agents are sent to the collector then forwarded via the internet to an Elastic endpoint.
+It is possible to opt-out by configuring the `agent_metrics` stanza in the collector configuration YAML file.
+When opting-out, troubleshooting Universal Profiling agent issues by our customer service will require extracting and providing the telemetry data manually.
```
Comment (bmorelli25) [inline docs/en/observability/profiling-self-managed-ops.asciidoc:185]:
```suggestion
By default, telemetry data collected by all profiling agents is sent to the collector, which then forwards it to an Elastic endpoint over the internet.
You can opt out by configuring the `agent_metrics` stanza in the collector configuration YAML file.
If you opt out, any troubleshooting by customer service will require manually extracting and providing the telemetry data.
```

File: docs/en/observability/profiling-self-managed-ops.asciidoc [hunk 6]
```diff
@@ -172,15 +172,81 @@ This endpoint is configured through the application's `host` configuration optio
...
+
+The content of the <<profiling-self-managed-running-linux-configfile-collector, "Collector configuration file">>
+provides a way to configure if telemetry data should be forwarded to Elastic, collected internally, or discarded.
+If you are running the collector in a network that has no internet access, the telemetry data will not be forwarded to Elastic.
```
Comment (bmorelli25) [inline docs/en/observability/profiling-self-managed-ops.asciidoc:189]:
```suggestion
The <<profiling-self-managed-running-linux-configfile-collector,"Collector configuration file">>
allows you to configure whether telemetry data is forwarded to Elastic, stored internally, or discarded.
If the collector is deployed in a network without internet access, telemetry data will not be forwarded to Elastic.
```

File: docs/en/observability/profiling-self-managed-ops.asciidoc [hunk 7]
```diff
@@ -172,15 +172,81 @@ This endpoint is configured through the application's `host` configuration optio
...
+
+Below are some examples of configurations to adopt for each case.
```
Comment (bmorelli25) [inline docs/en/observability/profiling-self-managed-ops.asciidoc:191]:
```suggestion
Below are examples of configurations for each option:
```

File: docs/en/observability/profiling-self-managed-ops.asciidoc [hunk 8]
```diff
@@ -172,15 +172,81 @@ This endpoint is configured through the application's `host` configuration optio
...
+
+**Forward telemetry data to Elastic**
+
+This is the default configuration.
+When the `agent_metrics` stanza is not present in the collector configuration file, the collector forwards telemetry data to Elastic.
+
+Enabling it explicitly has no difference in behavior.
```
Comment (bmorelli25) [inline docs/en/observability/profiling-self-managed-ops.asciidoc:198]:
```suggestion
This is the default configuration.
When the `agent_metrics` stanza is absent from the collector configuration file, the collector forwards telemetry data to Elastic.

Explicitly enabling it does not alter the default behavior.
```

File: docs/en/observability/profiling-self-managed-ops.asciidoc [hunk 9]
```diff
@@ -172,15 +172,81 @@ This endpoint is configured through the application's `host` configuration optio
...
+----
+agent_metrics:
+  disabled: false
+----
+
+**Collect telemetry data internally as well as send them to Elastic**
+
+In this configuration the same telemetry data gathered by the profiling agent are sent to Elastic and stored internally.
+The telemetry data are stored in the `profiling-metrics*` indices in the same Elasticsearch cluster storing Universal Profiling data,
+and they are subject to the same data retention policies.
```
Comment (bmorelli25) [inline docs/en/observability/profiling-self-managed-ops.asciidoc:210]:
```suggestion
**Collect telemetry data internally and send it to Elastic**

In this configuration, telemetry data from profiling agents is sent to Elastic **and** stored internally.
The data is stored in the `profiling-metrics*` indices within the same Elasticsearch cluster as Universal Profiling data
and follows the same data retention policies.
```

File: docs/en/observability/profiling-self-managed-ops.asciidoc [hunk 10]
```diff
@@ -172,15 +172,81 @@ This endpoint is configured through the application's `host` configuration optio
...
+agent_metrics:
+  disabled: false
+  write_all: true
+----
+
+**Collect telemetry data only internally**
+
+To collect telemetry data but not forward it to Elastic, configure the collector to store the telemetry data internally.
+It is possible to customize the Elasticsearch deployment storing the telemetry data by providing a list of Elasticsearch
+hosts and an API key to authenticate the requests.
```
Comment (bmorelli25) [inline docs/en/observability/profiling-self-managed-ops.asciidoc:223]:
```suggestion
**Collect telemetry data only internally**

To collect telemetry data without forwarding it to Elastic, configure the collector to store the data internally.
You can specify the Elasticsearch deployment for storing telemetry data by providing a list of Elasticsearch
hosts and an API key for authentication.
```

**Review by bmorelli25 — COMMENTED** [review]
Thanks! I have a few suggestions.

File: docs/en/observability/profiling-self-managed-ops.asciidoc (hunk 9 above)
Comment (inge4pres) [reply docs/en/observability/profiling-self-managed-ops.asciidoc:210]: so much clearer... thanks

**Comment by inge4pres** [comment]
Tanks @bmorelli25, reviewed and applied all suggestions, they were very good 👍🏼

**Review by bmorelli25 — APPROVED** [review]

**Pull request merged**